rfcs/experimental/flow_graph_wait_try_put_and_wait/README.md

```diff
@@ -0,0 +1,510 @@
+
+### Buffering the metainfo
+
+As described in the [buffering nodes description section](#single-push-buffering-nodes), current proposal requires the nodes that stores the user values as part of buffers to store
+also the metainformation associated with these values in the same buffer for future use and also to hold the additional reference counter to force the `try_put_and_wait` to wait 
+until the rejecting receiver take the item from the buffer and process it. 
+
+Such a behavior may significantly affect the other common use-case for buffering nodes: when they are used to store the result of the computation at the end of the Flow Graph. 
+In such scenarios, the metainformation would be stored in the buffer and never taken from it since the buffering node is not used as part of push-pull protocol and hence there is
+no rejecting successor. The `try_put_and_wait` function associated with such a metainformation will hang forever since the reference counter on the stored metainformation would
+never be decreased.
+
+It is impossible to rely on the number of successors while making a decision to store the metainformation in the buffer since if the node is used as part of the push-pull protocol,
+the number of successors is also equal to `0` since the edge is reversed.
+
+Current proposal considers this scenario is a `try_put_and_wait` feature limitation and does not add any support for that.
+
+### Multi-output nodes support
+
+Multi-output nodes (`multifunction_node` and `async_node`) creates an extra issue for the wait-for-one feature. Consider the following example (all of the examples shown
+in this section are for `multifunction_node` but also affects `async_node` in the same manner):
+
+```cpp
+
+using mf_node_type = multifunction_node<int, std::tuple<int, int>>;
+using output_ports = typename mf_node_type::output_ports_type;
+
+mf_node_type mfn(g, unlimited,
+    [](int input, output_ports& ports) {
+        std::get<0>(ports).try_put(output);
+        std::get<1>(ports).try_put(output);
+    })
+
+```
+
+Unlike the `function_node`, the computed output is not propagated implicitly to the successors. It is done explicitly by the user by calling
+`try_put(output)` on the corresponding output_port of the node. Since the metainformation associated with the particular input is hidden from the user
+inside of the Flow Graph implementation, it cannot be propagated to the successors as part of the explicit user try_put.
+
+An other interesting use-case is when the `multifunction_node` is used as a reduction for multiple input values with only one output. In that case the metainformation
+should not be automatically propagated at all and should be accumulated simultaneously with the outputs.
+
+Current implementation does not support ``multifunction_node`` and ``async_node``. 
+
+Possible approaches for implementing such support:
```

Comment:
Would all three approaches work for async node as well?

Are we not going to even consider an approach where the metainfo is passed and propagated with the message through the whole graph? Essentially the second option in the list, but applied to all node types?